services/scabbard/libscabbard/src/client/mod.rs

```diff
@@ -29,18 +29,69 @@ use super::protocol::SCABBARD_PROTOCOL_VERSION;
 
 pub use error::ScabbardClientError;
 
+/// Builder for building a ScabbardClient.
+#[derive(Default)]
+pub struct ScabbardClientBuilder {
+    url: Option<String>,
+    #[cfg(feature = "client-auth")]
+    auth: Option<String>,
+}
+
+impl ScabbardClientBuilder {
+    /// Creates a new ScabbardClientBuilder.
+    pub fn new() -> Self {
+        ScabbardClientBuilder::default()
+    }
+
+    /// Sets the `url` field of the ScabbardClientBuilder. The url will be used
+    /// as the bind endpoint for the Splinter REST API.
+    ///
+    /// # Arguments
+    ///
+    /// * `url` - The URL of the bind endpoint of the Splinter REST API.
+    pub fn with_url(mut self, url: &str) -> Self {
+        self.url = Some(url.into());
+        self
+    }
+    /// Sets the `auth` field of the ScabbardClientBuilder. The `auth` string contains a
+    /// signed json web token and will be passed to the Splinter REST API in an
+    /// Authorization header.
+    ///
+    /// # Arguments
+    ///
+    /// * `auth` - The authorization string containing a signed json web token.
+    #[cfg(feature = "client-auth")]
+    pub fn with_auth(mut self, auth: &str) -> Self {
+        self.auth = Some(auth.into());
+        self
+    }
+    /// Builds a ScabbardClient.
+    ///
+    /// # Errors
+    ///
+    /// * Returns an error if url is not set
+    /// * Returns an error if the client-auth feature is enabled and auth is not set
+    pub fn build(self) -> Result<ScabbardClient, ScabbardClientError> {
+        Ok(ScabbardClient {
+            url: self.url.ok_or_else(|| {
+                ScabbardClientError::new("Failed to build client, url not provided")
+            })?,
+            #[cfg(feature = "client-auth")]
+            auth: self.auth.ok_or_else(|| {
+                ScabbardClientError::new("Failed to build client, jwt authorization not provided")
+            })?,
+        })
+    }
+}
+
 /// A client that can be used to interact with scabbard services on a Splinter node.
 pub struct ScabbardClient {
     url: String,
+    #[cfg(feature = "client-auth")]
```

Comment:
This change should be in a separate commit, since it's distinct from the addition of a builder.

Reply:
Moved this to its own commit